spring-integration-amqp/src/main/java/org/springframework/integration/amqp/inbound/AmqpInboundChannelAdapter.java

```diff
@@ -123,13 +159,116 @@ public int beforeShutdown() {
 	}
 
 
-	/**
-	 * {@inheritDoc}
-	 * <p>No-op
-	 */
 	@Override
 	public int afterShutdown() {
 		return 0;
 	}
 
+	/**
+	 * If there's a retry template, it will set the attributes holder via the listener. If
+	 * there's no retry template, but there's an error channel, we create a new attributes
+	 * holder here. If an attributes holder exists (by either method), we set the
+	 * attributes for use by the {@link ErrorMessageStrategy}.
+	 * @param record the record.
```

Comment:
![image](https://cloud.githubusercontent.com/assets/1071590/26799006/bd4ba8e2-4a01-11e7-9bbd-037a11c264e0.png)

So, should be `amqpMessage`